app/records/api.py

```diff
@@ -20,16 +40,73 @@ def record_details_by_id(id, params={}) -> Record:
     raise ResourceNotFound(f"id {id} does not exist")
 
 
-def record_details_by_ref(reference, params={}):
+def record_details_by_ref(reference: str, params: dict = {}):
+    """
+    Fetches a record by its reference number.
+
+    Args:
+        reference: The reference number to search for
+        params: Optional additional parameters for the API request
+
+    Note:
+        TODO: Implement once Rosetta has support for reference-based lookup
+    """
     # TODO: Implement record_details_by_ref once Rosetta has support
     pass
 
 
-def wagtail_request_handler(uri, params={}) -> dict:
-    """Prepares and initiates Wagtail API requests using JSONAPIClient"""
+def wagtail_request_handler(uri: str, params: dict = {}) -> dict:
+    """
+    Prepares and initiates Wagtail API requests using JSONAPIClient.
```

Comment:
Could the `wagtail_request_handler` moved to `app/lib/api.py` where the other handlers are?